### PR TITLE
Exposes the receivedData signal

### PR DESCRIPTION
**Pull request by aheck**
This is quite useful to read data on the terminal and react to it (similar to what expect does)

### CODE REVIEW

File: lib/qtermwidget.cpp [hunk 1]
```diff
@@ -89,6 +89,8 @@ Session *TermWidgetImpl::createSession(QWidget* parent)
     session->setHistoryType(HistoryTypeBuffer(1000));
 
     session->setDarkBackground(true);
+    QObject::connect(session, SIGNAL(receivedData(QString)),
+            parent, SIGNAL(receivedData(QString)));
```
Comment (yan12125) [inline lib/qtermwidget.cpp]: Could you change this to Qt5's style instead of Qt4's SIGNAL() & SLOT() style?

**Comment by aheck** [comment]
I updated it. I don't like the static cast but it seems necessary.

**Comment by yan12125** [comment]
I guess the cast is not necessary, but I can't say without giving it a try.

**Comment by aheck** [comment]
The compiler refuses to compile because it can't match QTermWidget::receivedData on a pointer of type QWidget. This type cast works only as long as parent is a QTermWidget. Which should always be the case.

Alternatively and to make sure I could check the type of parent by asking the meta object system for the real classname and only execute the connect code with the typecast if parent really is a QTermWidget.

**Comment by yan12125** [comment]
Is it possible to put the ```connect``` call in ```QTermWidget::init```?

**Comment by aheck** [comment]
Good idea, that worked.

**Comment by yan12125** [comment]
Thanks. That looks good. Could you move the ```connect``` call and put it together with other ```connect``` calls? And use just connect() instead of QObject::connect to make codes shorter.

**Comment by aheck** [comment]
I updated it

**Comment by yan12125** [comment]
Sorry, but seems there's a bug in Github that breaks this pull request - the downloaded patch contains both setAutoClose and receivedData. I guess merging all changes into a single commit and force-push to this pull request works. Could you try it?

**Comment by aheck** [comment]
Should look better now

**Comment by yan12125** [comment]
Thanks!

**Pull request merged**